### PR TITLE
make close work as an inline close instead of a asynchronous close

### PR DESCRIPTION
**Pull request by paetling**
**Context**
Implements idea described here: https://github.com/propelinc/freshebt-vue/pull/346

### CODE REVIEW

**Comment by paetling** [comment]
@rammie What should I be merging against? I am very unclear where this should go

File: www/localtunnel.js [hunk 1]
```diff
@@ -46,7 +46,12 @@
             exec(success, error, 'LocalTunnel', 'getAllCookies', [url]);
         },
         close: function (eventname) {
-            exec(null, null, 'LocalTunnel', 'close', []);
+            return new Promise((resolve) => {
+                // NOTE(Alex) setTimeout seems to be needed to yeild the thread so that the browser
```
Comment (rammie) [inline www/localtunnel.js]: nit: yeild -> yield

**Review by rammie — APPROVED** [review]
Nice work!

File: www/localtunnel.js [hunk 2]
```diff
@@ -46,7 +46,12 @@
...
+                // can be fully cleaned up
+                this.addEventListener('exit', () => setTimeout(resolve, 0));
```
Comment (rammie) [inline www/localtunnel.js]: On second glace, a couple of things come to mind. (1) I'm not certain about arrow functions being transpiled correctly by cordova. (2) You might want to remove the event listener after it is invoked. Something like:

```javascript
var handler = (function() {
  this.removeEventListener('exit', handler);
  setTimeout(resolve, 0);
}).bind(this);
this.addEventListener('exit', handler);
```

**Pull request merged**